src/crypto/CryptState.h

```diff
@@ -7,21 +7,45 @@
 #define MUMBLE_CRYPTSTATE_H_
 
 #include "Timer.h"
+#include <chrono>
+#include <queue>
 #include <string>
 
+struct PacketStats {
+	unsigned int good   = 0;
+	unsigned int late   = 0;
+	unsigned int lost   = 0;
+	unsigned int resync = 0;
+};
+
+struct PacketStatsSnapshot {
+	PacketStats stats;
+	std::chrono::time_point< std::chrono::steady_clock > timestamp;
+};
+
 class CryptState {
 private:
 	Q_DISABLE_COPY(CryptState)
+
+	const std::chrono::seconds m_rollingScanInterval = std::chrono::seconds(5);
+	std::chrono::time_point< std::chrono::steady_clock > m_rollingTimer =
+		std::chrono::time_point< std::chrono::steady_clock >();
```

Comment:
Imo, something that is not a timer object, should not be called "timer". This makes code a bit confusing
```suggestion
	std::chrono::time_point< std::chrono::steady_clock > m_lastStatsSampleTime = {};
```